rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java

```diff
@@ -446,12 +446,20 @@ private MavenRepository applyAuthenticationToRepository(MavenRepository reposito
      * Returns a request builder with Authorization header set if the provided repository specifies credentials
      */
     private HttpSender.Request.Builder applyAuthenticationToRequest(MavenRepository repository, HttpSender.Request.Builder request) {
-        if (repository.getUsername() != null && repository.getPassword() != null) {
+        if (isViableCredential(repository.getUsername()) && isViableCredential(repository.getPassword())) {
             return request.withBasicAuthentication(repository.getUsername(), repository.getPassword());
         }
         return request;
     }
 
+    private boolean isViableCredential(@Nullable String value) {
+        return value != null && !isUnresolvedProperty(value);
+    }
+
+    private boolean isUnresolvedProperty(String value) {
+        return value.startsWith("${") && value.endsWith("}");
+    }
```

Comment:
Is it possible for us to resolve any property placeholders rather than silently discarding them?

Reply:
Sorta -- if I'm tracing correctly, then any resolvable properties would have already been resolved by this point.

Property interpolation is handled by either:
* The `MavenExecutionRequest` from the invoker library (in the case of the `rewrite-maven-plugin`), which resolves placeholders before our code even gets ahold of it
* Our `MavenXmlMapper`, per the functionality introduced over here: #1955

So if a placeholder is still an unresolved by the time we get here, then it *cannot* be resolved.

If you see a route to this code which doesn't depend on the two cases above, then yeah, we could borrow some of the logic from #1955 to attempt interpolation here, too.